### PR TITLE
fix: safer ends_with_semicolon with token errors

### PR DESCRIPTION
**Pull request by mscolnick**
This pull request improves the robustness of the `ends_with_semicolon` function in the `marimo/_ast/compiler.py` module by handling tokenization errors and adds comprehensive unit tests to verify its behavior across various edge cases.

### CODE REVIEW

File: tests/_ast/test_compiler.py [hunk 1]
```diff
@@ -444,6 +444,57 @@ def f() -> None:
...
+        assert compiler.ends_with_semicolon("   ") is False
+
+    @staticmethod
+    def test_token_error_unterminated_string() -> None:
+        # Unterminated string causes TokenError; fallback should handle it
+        assert compiler.ends_with_semicolon('x = "unterminated') is False
+
+    @staticmethod
+    def test_token_error_unterminated_string_with_semicolon() -> None:
+        assert compiler.ends_with_semicolon('x = "unterminated;') is True
```
Comment (dmadisetti) [inline tests/_ast/test_compiler.py:487]: Oh brutal
